File: map-service/src/main/java/com/example/map/repository/CircleRepository.java
package com.example.map.repository;

import com.example.map.entity.Circle;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface CircleRepository extends JpaRepository<Circle, Long> {

    List<Circle> findByOwnerId(Long ownerId);
}

File: apps/map-service/src/circle/circle.repository.ts
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { Circle } from './circle.entity';

@Injectable()
export class CircleRepository {
  constructor(
    @InjectRepository(Circle)
    private readonly repo: Repository<Circle>,
  ) {}

  findByOwnerId(ownerId: number): Promise<Circle[]> {
    return this.repo.find({
      where: { ownerId },
      relations: ['members'],
    });
  }

  save(circle: Circle): Promise<Circle> {
    return this.repo.save(circle);
  }
}
